code/game/objects/items/wayfinding.dm

```diff
@@ -1,32 +1,138 @@
+		set_expression("unsure", 2 SECONDS)
+		say("<span class='robot'>I know you already have a pinpointer, [user.first_name()].</span>")
+		user_spawn_cooldowns[user.real_name] = world.time + spawn_cooldown //spawn timer resets for trickers
+		return
+
+	var/msg
+	var/dispense = TRUE
+	var/obj/item/pinpointer/wayfinding/pointat
+	for(var/obj/item/pinpointer/wayfinding/WP in range(7, user))
+		if(WP.Adjacent(user))
+			set_expression("facepalm", 2 SECONDS)
+			say("<span class='robot'>[WP.owner == user.real_name ? "Your" : "A"] pinpointer is right there.</span>")
+			pointat(WP)
+			user_spawn_cooldowns[user.real_name] = world.time + spawn_cooldown
+			return
+		else if(WP in oview(7, user))
+			pointat = WP
+			break
+
+	if(world.time < user_spawn_cooldowns[user.real_name])
+		var/secsleft = (user_spawn_cooldowns[user.real_name] - world.time) / 10
+		msg += "to wait another [secsleft/60 > 1 ? "[round(secsleft/60,1)] minute\s" : "[round(secsleft)] second\s"]"
+		dispense = FALSE
+
+	var/datum/bank_account/cust_acc = null
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.get_bank_account())
+			cust_acc = H.get_bank_account()
+
+	if(cust_acc)
+		if(!cust_acc.has_money(ppt_cost))
+			msg += "[!msg ? "to find [ppt_cost-cust_acc.account_balance] more credit\s" : " and find [ppt_cost-cust_acc.account_balance] more credit\s"]"
+			dispense = FALSE
+
+	if(!dispense)
+		set_expression("sad", 2 SECONDS)
+		if(pointat)
+			msg += ". I suggest you get [pointat.owner == user.real_name ? "your" : "that"] pinpointer over there instead"
+			pointat(pointat)
+		say("<span class='robot'>You will need [msg], [user.first_name()].</span>")
 		return
 
-	to_chat(user, "<span class='notice'>You take a pinpointer from [src].</span>")
+	if(synth_acc.transfer_money(cust_acc, ppt_cost))
+		set_expression("veryhappy", 2 SECONDS)
+		say("<span class='robot'>That is [ppt_cost] credits. Here is your pinpointer.</span>")
+		var/obj/item/pinpointer/wayfinding/P = new /obj/item/pinpointer/wayfinding(get_turf(src))
+		user_spawn_cooldowns[user.real_name] = world.time + spawn_cooldown
+		user.put_in_hands(P)
+		P.owner = user.real_name
+
+/obj/machinery/pinpointer_dispenser/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/pinpointer/wayfinding))
+		var/obj/item/pinpointer/wayfinding/WP = I
+		to_chat(user, "<span class='notice'>You put \the [WP] in the return slot.</span>")
+		var/rfnd_amt
+		if(synth_acc.has_money(TRUE) && !WP.roundstart || WP.owner != user.real_name)
+			if(synth_acc.has_money(refund_amt))
+				rfnd_amt = refund_amt
+			else
+				rfnd_amt = synth_acc.account_balance
+			synth_acc._adjust_money(-rfnd_amt)
+			var/obj/item/holochip/HC = new /obj/item/holochip(user.loc)
+			HC.credits = rfnd_amt
+			HC.name = "[HC.credits] credit holochip"
+			if(istype(user, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = user
+				if(!H.put_in_hands(HC))
+					H.pulling = HC
+			else
+				user.pulling = HC
```

Comment:
Just drop it, dont set pulling. If they want to pull it, they can do so themselves.